### ISSUE TITLE
a question about CPC you mentioned in your paper

### ISSUE THREAD

**Issue by ql1074754250** [charlesCXK/TorchSemiSeg#75]
Hi, a nice work!
However, I have a question about CPC（Cross probability consistency）.
![image](https://user-images.githubusercontent.com/33171771/198509919-86770e24-819c-4b7f-b286-a2cbcbda5e1c.png)
The results of the first and second calculations should be the same. Why should you keep the two?

Thanks.

**Comment by mai556** [comment]
If you use kl-div as l_2, it will be not the same.

**Comment by charlesCXK** [comment]
When we use l2 loss in PyTorch, e.g., loss = MSE(x, y), y will have no gradient.
